emgapi/models.py

```diff
@@ -79,35 +82,81 @@ class Reason(models.IntegerChoices):
         TEMPORARY_SUPPRESSED = 7
         TEMPORARY_KILLED = 8
 
+        ANCESTOR_SUPPRESSED = 100
+
     is_suppressed = models.BooleanField(db_column='IS_SUPPRESSED', default=False)
     suppressed_at = models.DateTimeField(db_column='SUPPRESSED_AT', blank=True, null=True)
     suppression_reason = models.IntegerField(db_column='SUPPRESSION_REASON', blank=True, null=True, choices=Reason.choices)
 
-    def suppress(self, suppression_reason=None, save=True):
+    def suppress(self, suppression_reason=None, save=True, propagate=True):
         self.is_suppressed = True
         self.suppressed_at = timezone.now()
         self.suppression_reason = suppression_reason
         if save:
             self.save()
+        if propagate:
+            for descendant_relation in self.suppressible_descendants:
+                descendants_to_suppress: QuerySet = getattr(
+                    self,
+                    descendant_relation
+                ).filter(
+                    is_suppressed=False
+                )
+                for descendant in descendants_to_suppress:
+                    descendant.is_suppressed = True
+                    descendant.suppression_reason = self.Reason.ANCESTOR_SUPPRESSED
+                descendants_to_suppress.bulk_update(
+                    descendants_to_suppress,
+                    [
+                        'is_suppressed',
+                        'suppression_reason'
+                    ]
+                )
+                logger.info(
+                    f'Propagated suppression of {self} '
+                    f'to {len(descendants_to_suppress)} {descendant_relation} descendants'
+                )
         return self
 
-    def unsuppress(self, suppression_reason=None, save=True):
+    def unsuppress(self, suppression_reason=None, save=True, propagate=True):
         self.is_suppressed = False
         self.suppressed_at = None
         self.suppression_reason = None
         if save:
             self.save()
+        if propagate:
```

Comment:
@SandyRogers this might be overkill, and please ignore if it's going to cause an uneccessary delay, 
but I think in the long term, it will be better to refactor the suppress and unsuppress methods.
It seems like they are doing the same thing, with only slight variations. 
So it might be better if they both call a generalised method, but with different arguments.